packages/rpc/README.md

```diff
@@ -14,4 +14,108 @@
 
 # @solana/rpc
 
-TODO
+This package contains utilities for creating objects that you can use to communicate with a Solana JSON RPC server. It can be used standalone, but it is also exported as part of the Solana JavaScript SDK [`@solana/web3.js@experimental`](https://github.com/solana-labs/solana-web3.js/tree/master/packages/library).
+
+## Types
+
+### `RpcTransport{Devnet|Testnet|Mainnet}`
+
+These types refine the base `RpcTransport` type. Each describes a transport that is specific in some way to a particular Solana cluster.
+
+For instance, a `RpcTransportDevnet` is understood to communicate with a RPC server related to devnet, and as such might only be accepted for use as the transport of a `RpcDevnet`.
+
+This is useful in cases where you need to make assertions about what capabilities a RPC offers. For example, RPC methods like `requestAirdrop` are not available on mainnet. You can use the ability to assert on the type of RPC transport at compile time to prevent calling unimplemented methods or presuming the existence of unavailable capabilities.
+
+### `RpcTransportFromClusterUrl<TClusterUrl extends ClusterUrl>`
+
+Given a `ClusterUrl`, this utility type will resolve to as specific a `RpcTransport` as possible.
+
+```ts
+function createCustomTransport<TClusterUrl extends ClusterUrl>(
+    clusterUrl: TClusterUrl,
+): RpcTransportFromClusterUrl<TClusterUrl> {
+    /* ... */
+}
+const transport = createCustomTransport(testnet('http://api.testnet.solana.com'));
+transport satisfies RpcTransportTestnet; // OK
+```
+
+### `Rpc{Devnet|Testnet|Mainnet}<TRpcMethods>`
+
+These types refine the base `Rpc` type. Each describes a RPC that is specific in some way to a particular Solana cluster and a corpus of RPC methods.
+
+This is useful in cases where you need to make assertions about the suitablilty of a RPC for a given purpose. For example, you might like to make it a type error to combine certain types with RPC belonging to certain clusters, at compile time.
+
+```ts
+async function getSpecialAccountInfo(
+    address: Address<'ReAL1111111111111111111111111111'>,
+    rpc: RpcMainnet,
+): Promise<SpecialAccountInfo>;
+async function getSpecialAccountInfo(
+    address: Address<'TeST1111111111111111111111111111'>,
+    rpc: RpcDevnet | RpcTestnet,
+): Promise<SpecialAccountInfo>;
+async function getSpecialAccountInfo(address: Address, rpc: Rpc): Promise<SpecialAccountInfo> {
+    /* ... */
+}
+const rpc = getSolanaRpc({ url: devnet('https://api.devnet.solana.com') });
+await getSpecialAccountInfo(address('TeST1111111111111111111111111111'), rpc); // ERROR
```

Comment:
According to your type signatures above this is valid. Maybe you meant to use the "real" address here which would cause a type failure.

Reply:
I did! Oops.